doc/docs/master/concepts/pipeline-concepts/pipeline/spout.md

```diff
@@ -1,111 +1,132 @@
-errors produced by closing a previous write to the pipe, retries the write
-operation.
-1. Closes the `tar` stream and connection to `/pfs/out`, which produces the
-commit.
+A spout pipeline differs
+from regular pipelines in many ways:
+
+- its code runs continuously, waiting for new events.
+- it does not take an input repo. 
+Instead, it consumes data from an outside source.
+- the pipeline's output repo,
+`pfs/out` is not directly accessible. 
+To write into the output repo,
+you will need to use the `put file` API call
+via any of the following:
+    - the CLI (`pachctl put file`)
+    - one of the Pachyderm's SDKs (for [golang](https://docs.pachyderm.com/latest/reference/clients/#go-client) or [Python](https://docs.pachyderm.com/latest/reference/clients/#python-client) )
+    - or [your own API client](https://docs.pachyderm.com/latest/reference/clients/#other-languages).
+- Pachyderm CLI (`pachctl`) is packaged
+in the base image of your spout
+as well as your authentication information.
+As a result, the authentication is seamless when using `pachctl`. 
+
+!!! Note 
+    Support for a transparent
+    authentication in our SDKs is coming soon.
+    In the meantime, check our Spout 101 example
+    at the end of this page
+    to learn how to retrieve
+    and inject your authentication token
+    into your API client.
+
+
+To create a spout pipeline, you will need:
+
+* A source of streaming data.
+* A Docker container with your spout code that connects to, reads, transforms, and pushes data from the data source to your output repo. 
+* A spout pipeline specification file that uses your container.
+
+!!! Note 
+    It is important
+    to remember that you will
+    need to use a `put file` API call
+    from a client of your choice
+    to push your data
+    into the pipeline output repository.
+    Having the entire Pachyderm API
+    available to you
+    allows you to package data
+    into commits and transactions
+    at the granularity your problem requires.
+
 
 A minimum spout specification must include the following
 parameters:
 
 | Parameter   | Description |
 | ----------- | ----------- |
-| `name`      | The name of your data pipeline and the output repository. You can set an <br> arbitrary name that is meaningful to the code you want to run. |
-| `transform` | Specifies the code that you want to run against your data, such as a Python <br> or Go script. Also, it specifies a Docker image that you want to use to run that script. |
-| `overwrite` | (Optional) Specifies whether to overwrite the existing content <br> of the file from previous commits or previous calls to the <br> `put file` command  within this commit. The default value is `false`. |
+| `name`      | The name of your data pipeline and the output repository. You can set an arbitrary name that is meaningful to the code you want to run. |
+| `spout`      | This section can be left empty. Optional: Add a service field to expose your spout as a service. |
```

Comment:
See prior comment about referring to the precise syntax.  I've been using the word "attribute" for a parameterized field added to a toplevel field, but I'm not sure that's completely correct.  It would be worth making sure we're using the exact terminology for these things that JSON and YAML use.

Reply:
Attribute
     |- Field
works. 
I have not found a norm/naming convention but do agree that we need to be consistent.